### PR TITLE
Modernize python to >= 3.6

### PR DESCRIPTION
**Pull request by klieret**
python 3.5 has reached its EOL, so let's explicitly require >= 3.6 and make use of its features

### CODE REVIEW

**Comment by DavidMStraub** [comment]
Cool, did you use a script for this?

**Comment by klieret** [comment]
Yes `pyupgrade`. I can highly recommend it! It has a couple of switches for which python version to use. I usually use it as a pre-commit hook

https://github.com/asottile/pyupgrade

**Comment by klieret** [comment]
There's only one file here where `ast` cannot parse the python: `./wilson/match/smeft_loop.py`. It runs into this:

```
RecursionError: maximum recursion depth exceeded while calling a Python object
```

Not sure if that tells us that something problematic is happening in this file or that the formulas there are too complex.

**Comment by klieret** [comment]
It's probably just too long for one expression. [This](https://bugs.python.org/issue32758) is an old issue that threw similar exceptions and SO has a couple of similar posts, like [this one](https://stackoverflow.com/questions/30107087/runtimeerror-maximum-recursion-depth-exceeded-while-calling-a-python-object-wit)

**Comment by dvandyk** [comment]
> There's only one file here where `ast` cannot parse the python: `./wilson/match/smeft_loop.py`. It runs into this:
> 
> ```
> RecursionError: maximum recursion depth exceeded while calling a Python object
> ```
> 
> Not sure if that tells us that something problematic is happening in this file or that the formulas there are too complex.

The formulas are excessively long, and this code is a bottleneck for the speed of evaluation. At some point we should discuss cythonizing it.

File: wilson/wcxf/__init__.py [hunk 1]
```diff
@@ -14,9 +14,9 @@
 child_bases = glob.glob(os.path.join(_root, 'bases', 'child', '*.basis.json'))
 
 for eft in all_efts:
-    with open(eft, 'r') as f:
+    with open(eft) as f:
```
Comment (dvandyk) [inline wilson/wcxf/__init__.py:17]: Is this really a good idea, to be less specific in how you open a file? I see that pyupgrade removes 'U' and 'r', which are the default now. What if this changes in a future Python version? If I'm overlooking anything, feel free to rebase and merge :-).

**Review by dvandyk — APPROVED** [review]
Approved, but puzzled about the changes to calls to ``open``.

**Pull request merged**